nibabel/freesurfer/io.py

```diff
@@ -318,51 +318,81 @@ def read_annot(filepath, orig_ids=False):
+        # entries in the colour table. In new-format files, this will be
+        # equal to -2
         n_entries = np.fromfile(fobj, dt, 1)[0]
+
+        # We've got an old-format .annot file.
         if n_entries > 0:
+
+            # orig_tab string length + string
             length = np.fromfile(fobj, dt, 1)[0]
             orig_tab = np.fromfile(fobj, '>c', length)
             orig_tab = orig_tab[:-1]
-
             names = list()
-            ctab = np.zeros((n_entries, 5), np.int)
+            ctab = np.zeros((n_entries, 5), np.int32)
             for i in xrange(n_entries):
+                # structure name length + string
                 name_length = np.fromfile(fobj, dt, 1)[0]
                 name = np.fromfile(fobj, "|S%d" % name_length, 1)[0]
                 names.append(name)
+                # RGBA
                 ctab[i, :4] = np.fromfile(fobj, dt, 4)
-                ctab[i, 4] = (ctab[i, 0] + ctab[i, 1] * (2 ** 8) +
+                # generate the annotation value
+                ctab[i, 4] = (ctab[i, 0] +
+                              ctab[i, 1] * (2 ** 8) +
                               ctab[i, 2] * (2 ** 16) +
                               ctab[i, 3] * (2 ** 24))
+        # We've got a new-format .annot file
         else:
+            # file version number
             ctab_version = -n_entries
             if ctab_version != 2:
                 raise Exception('Color table version not supported')
-            n_entries = np.fromfile(fobj, dt, 1)[0]
-            ctab = np.zeros((n_entries, 5), np.int)
+            # maximum colour table index used
+            max_index = np.fromfile(fobj, dt, 1)[0]
+            ctab = np.zeros((max_index, 5), np.int32)
+            # orig_tab string length + string
             length = np.fromfile(fobj, dt, 1)[0]
             np.fromfile(fobj, "|S%d" % length, 1)[0]  # Orig table path
+            # number of entries to read from the file
             entries_to_read = np.fromfile(fobj, dt, 1)[0]
             names = list()
             for i in xrange(entries_to_read):
-                np.fromfile(fobj, dt, 1)[0]  # Structure
+                # index of this entry
+                idx = np.fromfile(fobj, dt, 1)[0]
+                # structure name length + string
                 name_length = np.fromfile(fobj, dt, 1)[0]
                 name = np.fromfile(fobj, "|S%d" % name_length, 1)[0]
                 names.append(name)
-                ctab[i, :4] = np.fromfile(fobj, dt, 4)
-                ctab[i, 4] = (ctab[i, 0] + ctab[i, 1] * (2 ** 8) +
-                              ctab[i, 2] * (2 ** 16))
-        ctab[:, 3] = 255
+                # RGBA
+                ctab[idx, :4] = np.fromfile(fobj, dt, 4)
+                ctab[idx, 4] = (ctab[idx, 0] +
+                                ctab[idx, 1] * (2 ** 8) +
+                                ctab[idx, 2] * (2 ** 16) +
+                                ctab[idx, 3] * (2 ** 24))
```

Comment:
Given this is now showing up a couple times, maybe it should be factored into something like:

```Python
# Above
def pack_rgba(ctab):
    # ctab :: n x 4
    bitshifts = 2 ** np.array([[0], [8], [16], [24]], dtype=ctab.dtype)
    return ctab.dot(bitshifts)
```

(If you find the manual expansion more readable than the vector math, go ahead and use that.)

Now, we can drop back two indentations, out of the for loop and the `if`/`else` and operate on slices:

```Python
# Construct annotation values
ctab[:, 4] = pack_rgba(ctab[:, :4])
```

(We can drop out of the `if`/`else`, since this appears at the end of both branches.)

Reply:
No problemo. I suppose I was trying to minimise code changes, despite the obvious potential for refactoring.